apps/backend/src/user/user.service.ts

```diff
@@ -0,0 +1,41 @@
+import { UserType } from '@animavita/models';
+import {
+  Inject,
+  Injectable,
+  NotFoundException,
+  UnprocessableEntityException,
+} from '@nestjs/common';
+
+import { UserRepository } from './entities/user-entities.interface';
+
+@Injectable()
+export class UserService {
+  constructor(
+    @Inject('MONGODB') private readonly userRepository: UserRepository,
+  ) {}
+
+  async create(user: UserType) {
+    const foundUser = await this.userRepository.findByEmail(user.email);
+
+    if (foundUser)
+      throw new UnprocessableEntityException('Email already taken');
+
+    return this.userRepository.create(user);
+  }
+
+  findById(userId: string) {
```

Comment:
Don't we need to implement a conditional for the case where the user is not found and then throw an error, as you did for `findById`?